src/reducers/index.js

```diff
@@ -2,9 +2,20 @@ import { combineReducers } from 'redux';
 import forms from './forms';
 import smData from './sm-data';
 import peaksInstance from './peaks-instance';
+import * as types from '../actions/types';
 
-export default combineReducers({
+const appReducer = combineReducers({
   forms,
   smData,
   peaksInstance
 });
+
+const rootReducer = (state, action) => {
+  if (action.type === types.RESET_STORE) {
+    // Reducers return initial state when they are called with 'undefined'
+    state = undefined;
```

Comment:
Yea, my only question is if this should be `state = {}` instead of an `undefined` value.

Reply:
The reducers set the state to initial state only when state is set to `undefined` (not declared/declared but has no value) not any other value. So the way to reset the state to its initial value is to pass `state = undefined` into the reducers.